src/libs/Navigation/AppNavigator/AppNavigator.js

```diff
@@ -0,0 +1,190 @@
+import _ from 'underscore';
+import React from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import {createStackNavigator, CardStyleInterpolators} from '@react-navigation/stack';
+import {createDrawerNavigator} from '@react-navigation/drawer';
+import createCustomModalStackNavigator from './createCustomModalStackNavigator';
+import AppWrapper from '../../../pages/home/AppWrapper';
+import styles, {
+    getNavigationDrawerType,
+    getNavigationDrawerStyle,
+    getNavigationModalCardStyle,
+} from '../../../styles/styles';
+import {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
+import modalRoutesPropTypes from './modalRoutesPropTypes';
+import mainRoutesPropTypes from './mainRoutesPropTypes';
+import sidebarRoutePropTypes from './sidebarRoutePropTypes';
+import publicRoutePropTypes from './publicRoutePropTypes';
+
+const RootStack = createStackNavigator();
+const Drawer = createDrawerNavigator();
+
+const propTypes = {
+    // If we have an authToken this is true
+    authenticated: PropTypes.bool.isRequired,
+
+    // Whem responsive is true we will use a custom navigator for our modals so they present in a custom way when at
+    // larger screen widths or on web and desktop platforms. Only native platforms with small screen widths use the
+    // react-navigation style modal presentation.
+    responsive: PropTypes.bool,
+
+    // Route object that provides the sidebar component we will render in the drawer navigator
+    sidebarRoute: sidebarRoutePropTypes.isRequired,
+
+    // The public route is the entry point for the unauthenticated app.
+    publicRoutes: PropTypes.arrayOf(publicRoutePropTypes).isRequired,
+
+    // The main routes display in the center on wide screens and non modally on small screens. There is only the report
+    // screen at the moment.
+    mainRoutes: mainRoutesPropTypes.isRequired,
+
+    // Modal routes are presented above the content or in a specified way on larger screen widths
+    modalRoutes: modalRoutesPropTypes.isRequired,
+
+    // Whether the drawer should be open on init
+    isDrawerOpenByDefault: PropTypes.bool.isRequired,
+
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    responsive: false,
+};
+
+const AppNavigator = (props) => {
+    const RootNavigator = RootStack;
+    return (
+        <RootNavigator.Navigator
+            mode="modal"
+        >
+            {props.authenticated
```

Comment:
This file could get big pretty fast once there's more stacks, I would separate these out into auth/unauth stacks. will help with deeplinking logic as well (will go more in detail below)

basically this file can simply be` {props.authenticated ? <AuthStackNavigator /> : <UnathStackNavigator />}`